tests/integration/api/test_activation_with_credential.py

```diff
@@ -346,3 +445,113 @@ def test_create_activation_with_conflict_credentials(
         "in extra var. It conflicts with credential type: type1. "
         "Please check injectors." in response.data["non_field_errors"]
     )
+
+
+@pytest.mark.django_db
+def test_create_activation_without_extra_vars(
+    client: APIClient,
+    kafka_credential_type: models.CredentialType,
+    preseed_credential_types,
+):
+    fks = create_activation_related_data(OVERLAP_EXTRA_VAR)
+    test_activation = TEST_ACTIVATION.copy()
+    test_activation.pop("extra_var_id")
+    test_activation["decision_environment_id"] = fks["decision_environment_id"]
+    test_activation["rulebook_id"] = fks["rulebook_id"]
+
+    credential_types = models.CredentialType.objects.bulk_create(
+        [
+            models.CredentialType(
+                name="user_type",
+                inputs={"fields": [{"a": "b"}]},
+                injectors={"extra_vars": {"username": "adam"}},
+            ),
+            models.CredentialType(
+                name="password_type",
+                inputs={"fields": [{"1": "2"}]},
```

Comment:
@hsong-rh we should fix the inputs here too to be sasl_username and sasl_password.